setup.py

```diff
@@ -9,18 +9,22 @@ def get_description():
     with open('README.rst', 'r') as f:
         return ''.join(f.readlines()[2:])
 
-requires = [
-    'fedmsg',
-    'moksha.hub>=1.4.6',
-    'fedmsg_meta_fedora_infrastructure',
-    'sqlalchemy',
-    'futures',
-]
-
-tests_require = [
-    'nose',
-    'freezegun'
-]
+def get_requirements(requirements_file='requirements.txt'):
+    """Get the contents of a file listing the requirements.
+
+    :arg requirements_file: path to a requirements file
+    :type requirements_file: string
+    :returns: the list of requirements, or an empty list if
+              `requirements_file` could not be opened or read
+    :return type: list
+    """
+
+    with open(requirements_file) as f:
+        return [
+            line.rstrip().split('#')[0]
+            for line in f.readlines()
+            if not line.startswith('#')
+        ]
```

Comment:
This is fine for the given list of dependencies, but note that it is possible to pip install from a git spec which can include a ```#``` to indicate which ref (branch, tag, etc.) to use. @jeremycline did something like this in Bodhi that does handle the git spec:

https://github.com/fedora-infra/bodhi/commit/03ac11177e631d26e7e8c36a224018a00bc6cdd9#diff-2eeaed663bd0d25b7e608891384b7298